tribits/common_tpls/FindTPLHDF5.cmake

```diff
@@ -2,78 +2,103 @@
 # See associated tribits/Copyright.txt file for copyright and license! #
 ########################################################################
 
-#
-# First, set up the variables for the (backward-compatible) TriBITS way of
-# finding HDF5.  These are used in case find_package(HDF5 ...) is not called
-# or does not find HDF5.  Also, these variables need to be non-null in order
-# to trigger the right behavior in the function
-# tribits_tpl_find_include_dirs_and_libraries().
-#
-
-set(REQUIRED_HEADERS hdf5.h)
-set(REQUIRED_LIBS_NAMES hdf5)
-
-if (HDF5_REQUIRE_FORTRAN)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} hdf5_fortran)
-endif()
+set(HDF5_INTERNAL_IS_MODERN FALSE)
 
-if (TPL_ENABLE_MPI)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} z)
-endif()
+if (Netcdf_ALLOW_MODERN)
+
+  set(minimum_modern_HDF5_version 1.13.2)
+  print_var(Netcdf_ALLOW_MODERN)
+  message("-- Using find_package(HDF5 ${minimum_modern_HDF5_version} CONFIG) ...")
+  find_package(HDF5  ${minimum_modern_HDF5_version}  CONFIG)
+  if (HDF5_FOUND)
```

Comment:
@ibaned and @gsjaardema, following on from the comments https://github.com/sandialabs/seacas/pull/336#pullrequestreview-1157178993, https://github.com/trilinos/Trilinos/pull/11175#discussion_r1006168735, and https://github.com/trilinos/Trilinos/pull/11175#discussion_r1006169418, is it worth considering if we should try to change the `find_package([HDF5|netCDF] <minVersion> CONFIG)` calls which require a specific version to `find_package([HDF5|netCDF] <minVersion>... CONFIG)` to only require a minimum version?

Wait, the book "Professional CMake" says:

> The optional version argument indicates that the package must be of the specified version or higher, or match exactly if `EXACT` is also given.

so it would seem that a single version number **is** supposed to already be interpreted as the minimum version, not the exact version.  (That is not at all clear from my reading of the [official find_package() documentation](https://cmake.org/cmake/help/v3.25/command/find_package.html).